### PR TITLE
[NPUW] Fixed Gemma2 4K sliding window work with short prompts on NPU

### PR DESCRIPTION
**Pull request by AsyaPronina**
### Details:
 - *Relaxed Phi3SlidingMask2 pattern in order to allow `attention_mask` and `position_ids` to be located more flexibly in the model without breaking the sliding window mask calculation*

### Tickets:
 - *EISW-190610*

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 1]
```diff
@@ -594,7 +594,7 @@ class Phi3SlidingMask2 : public ov::pass::MatcherPass {
 public:
     OPENVINO_MATCHER_PASS_RTTI("npuw::LLMCompiledModel::Phi3SlidingMask2");
 
-    Phi3SlidingMask2() {
+    Phi3SlidingMask2(const std::shared_ptr<ov::Model>& model) {
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: Normally we have `explicit` here, but what's the reason to pass model to a pass that can, in fact, be applied to any other model?

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 2]
```diff
@@ -690,14 +668,27 @@ class Phi3SlidingMask2 : public ov::pass::MatcherPass {
...
+            for (const auto& i : model->inputs()) {
+                if (i.get_any_name() == "position_ids") {
+                    matched_position_ids = i.get_node_shared_ptr();
+                }
+                if (i.get_any_name() == "attention_mask") {
+                    matched_attention_mask = i.get_node_shared_ptr();
+                }
+            }
+            OPENVINO_ASSERT(matched_position_ids, "position_ids input is not found!");
+            OPENVINO_ASSERT(matched_attention_mask, "attention_mask input is not found!");
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: Why can't you make it through pass? Or, maybe, outside the pass?

Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: Because it is long chain of operations till the `position_ids` and `attention_mask` starting from point of the pattern and this chain can differ between models.

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp [hunk 3]
```diff
@@ -837,10 +849,13 @@ void patch_phi3_sliding_mask(const std::shared_ptr<ov::Model>& model) {
     //        Qwen2.5 VL/Omni uses 3D position_ids, which can't be directly used
     //        in creation of sliding window mask.
     if (!ov::npuw::util::has_input(model, "token_type_ids") && !ov::npuw::util::has_input(model, "inputs_embeds")) {
-        ov::pass::GraphRewrite rewr;
-        rewr.add_matcher<Phi3SlidingMask2>();
-        rewr.add_matcher<Phi3SlidingMask>();
-        rewr.run_on_model(model);
+        ov::pass::Manager manager;
+        manager.register_pass<Phi3SlidingMask2>();
+        if (!manager.run_passes(model)) {
```
Comment (AlexanderKalistratov) [inline src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: I think it would be better to move this logic into `Phi3SlidingMask2` pass to make it handle all the cases.
And probably rename `Phi3SlidingMask2` into `Phi3SlidingMask` and `Phi3SlidingMask` rename into `Phi3SlidingMask_pattern_transformers_451` or something like that.

Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: Fixed, thanks!

File: src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp (hunk 1 above)
Comment (AsyaPronina) [reply src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp]: Fixed, thanks!

**Review by dmatveev — APPROVED** [review]
Thanks!
